src/theory/datatypes/project_op.h

```diff
@@ -0,0 +1,58 @@
+/*********************                                                        */
+/*! \file project_op.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mudathir Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief a class for MK_BAG operator
```

Comment:
Fix.

Reply:
Done. I wonder why MK_BAG is here. Perhaps this file was written in 2020 :)

Reply:
> > Also, for 0-argument projection, is the syntax `((_ project) t)` or should it just be `(project t)`? I'm not sure if SMT-LIB permits indexed symbols with 0 indices.
> 
> We have another issue as well. How do we represent an empty tuple. Below is the output of the regression
> 
> ```smt
> (declare-fun v () Tuple) ...
> ---------------------------------------------------------------------------------------------------------------------
> (
> ...
> (define-fun v () Tuple mkTuple)
> ...
> )
> ```
> 
> Cesare suggested something like
> 
> ```smt
> (define-fun v () Tuple emptyTuple) ; Unit data type has only one constructor
> ```

I think 0-argument tuples should use the same syntax as the tuple constructor, which is `mkTuple`, although we had discussed changing this to `tuple`.
Thus, I prefer the current implementation.